EngineLayer/CrosslinkSearch/CrosslinkSearchEngine.cs

```diff
@@ -515,14 +606,21 @@ private List<MatchedFragmentIon> ScoreChildScan(Ms2ScanWithSpecificMass parentSc
                     mainPeptide.OneBasedStartResidueInProtein, mainPeptide.OneBasedEndResidueInProtein,
                     mainPeptide.CleavageSpecificityForFdrCategory, mainPeptide.PeptideDescription,
                     mainPeptide.MissedCleavages, mod, mainPeptide.NumFixedMods);
-
-                peptideWithMod.Fragment(CommonParameters.ChildScanDissociationType, FragmentationTerminus.Both, childProducts);
+          
+                    peptideWithMod.Fragment(CommonParameters.MS3ChildScanDissociationType, FragmentationTerminus.Both, childProducts);              
             }
-            else if (Math.Abs(childScan.PrecursorMass - parentScan.PrecursorMass) < 0.01 && CommonParameters.DissociationType != CommonParameters.ChildScanDissociationType)
+            else if (Crosslinker.Cleavable && Math.Abs(childScan.PrecursorMass - parentScan.PrecursorMass) < 0.01)
```

Comment:
why hard coded mass differnce

Reply:
For example, for HCD-ETD type of data, it is better to make sure they have the same precursor mass.